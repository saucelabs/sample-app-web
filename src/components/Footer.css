.footer {
  background-color: #474c55;
  min-height: 360px;
  margin-top: auto;
  position: relative;
  /*display: inline-table;*/
}
.footer_copy,
.footer_robot,
.social {
  position: absolute;
}
.footer_copy {
  left: 50px;
  bottom: 40px;
  color: #fff;
}
.social {
  list-style-type: none;
  left: 10px;
  top: 20px;
}
.social li {
  float: left;
  margin-right: 20px;
  display: inline-block;
  width: 32px;
  height: 32px;
}
.social_twitter {
  background: url("../assets/img/twitter.png") no-repeat;
}
.social_facebook {
  background: url("../assets/img/facebook.png") no-repeat;
}
.social_linkedin {
  background: url("../assets/img/linkedIn.png") no-repeat;
}

.social a {
  width: 32px;
  height: 32px;
  display: block;
  font-size: 0;
}

.footer_robot {
  width: 470px;
  height: 352px;
  right: 0;
  bottom: 0;
}

@media only screen and (max-width: 960px) {
  .footer {
    height: auto;
    text-align: center;
    padding: 0 10px;
    display: inline-table;
  }
  .social,
  .footer_copy,
  .footer_robot {
    position: relative;
  }
  .social {
    width: 100%;
    padding: 0;
    left: 0;
  }
  .social li {
    float: none;
  }

  .footer_copy {
    width: 100%;
    left: 0;
    bottom: 0;
    margin: 40px 0;
  }
  .footer_robot {
    position: relative;
  }
  .footer_robot {
    width: 100%;
    height: 100%;
    max-width: 470px;
    max-height: 352px;
  }
}
