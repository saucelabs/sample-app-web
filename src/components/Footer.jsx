import React from "react";
import SwagBotFooter from "../assets/img/SwagBot_Footer_graphic.png";
import "./Footer.css";

const SwagLabsFooter = () => {
  return (
    <footer className="footer">
      <ul className="social">
        <li className="social_twitter">
          <a
            href="https://twitter.com/saucelabs"
            target="_blank"
            rel="noreferrer"
          >
            Twitter
          </a>
        </li>
        <li className="social_facebook">
          <a
            href="https://www.facebook.com/saucelabs"
            target="_blank"
            rel="noreferrer"
          >
            Facebook
          </a>
        </li>
        <li className="social_linkedin">
          <a
            href="https://www.linkedin.com/company/sauce-labs/"
            target="_blank"
            rel="noreferrer"
          >
            LinkedIn
          </a>
        </li>
      </ul>

      <div className="footer_copy">
        &copy; {new Date().getFullYear()} Sauce Labs. All Rights Reserved. Terms
        of Service | Privacy Policy
      </div>
      <img alt="Swag Bot Footer" className="footer_robot" src={SwagBotFooter} />
    </footer>
  );
};

export default SwagLabsFooter;
