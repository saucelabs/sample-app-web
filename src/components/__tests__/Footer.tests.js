import React from "react";
import { shallow } from "enzyme";
import Footer from "../Footer";

describe("Footer", () => {
  beforeEach(() => {
    jest.spyOn(Date.prototype, "getFullYear").mockReturnValue(1234);
  });

  it("should render correctly", () => {
    const component = shallow(<Footer />);

    expect(component).toMatchSnapshot();
  });
});
