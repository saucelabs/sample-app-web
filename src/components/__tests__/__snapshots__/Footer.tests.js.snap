// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<footer
  className="footer"
>
  <ul
    className="social"
  >
    <li
      className="social_twitter"
    >
      <a
        href="https://twitter.com/saucelabs"
        rel="noreferrer"
        target="_blank"
      >
        Twitter
      </a>
    </li>
    <li
      className="social_facebook"
    >
      <a
        href="https://www.facebook.com/saucelabs"
        rel="noreferrer"
        target="_blank"
      >
        Facebook
      </a>
    </li>
    <li
      className="social_linkedin"
    >
      <a
        href="https://www.linkedin.com/company/sauce-labs/"
        rel="noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
    </li>
  </ul>
  <div
    className="footer_copy"
  >
    Â© 
    1234
     Sauce Labs. All Rights Reserved. Terms of Service | Privacy Policy
  </div>
  <img
    alt="Swag Bot Footer"
    className="footer_robot"
    src="test-file-stub"
  />
</footer>
`;
